src/sentry/api/bases/organizationmember.py

```diff
@@ -41,19 +52,25 @@ def convert_args(self, request: Request, organization_slug, member_id="me", *arg
             except OrganizationMember.DoesNotExist:
                 raise ResourceDoesNotExist
 
-            return (args, kwargs)
+            return args, kwargs
         else:
             raise ResourceDoesNotExist
 
-    def _get_member(self, request: Request, organization, member_id):
+    def _get_member(
+        self, request: Request, organization: Organization, member_id: int | str
+    ) -> OrganizationMember:
         if member_id == "me":
             queryset = OrganizationMember.objects.filter(
                 organization=organization, user__id=request.user.id, user__is_active=True
             )
         else:
-            queryset = OrganizationMember.objects.filter(
-                Q(user__is_active=True) | Q(user__isnull=True),
-                organization=organization,
-                id=member_id,
-            )
+            try:
+                queryset = OrganizationMember.objects.filter(
+                    Q(user__is_active=True) | Q(user__isnull=True),
+                    organization=organization,
+                    id=member_id,
+                    invite_status=InviteStatus.APPROVED.value,
```

Comment:
@mgaeta this change to the query is separate to the reorganization. Can you move this change into a separate PR?